components/calendar/__tests__/index.test.tsx

```diff
@@ -86,6 +86,266 @@ describe('Calendar', () => {
+
+    fireEvent.click(container.querySelector('[title="2018-02-05"]')!);
+    fireEvent.click(container.querySelector('[title="2018-02-07"]')!);
+    expect(onSelect.mock.calls.length).toBe(2);
+
+    onSelect.mockReset();
+
+    openSelect(container, '.ant-picker-calendar-year-select');
+    fireEvent.click(Array.from(container.querySelectorAll('.ant-select-item-option')).at(2)!);
+    fireEvent.click(Array.from(container.querySelectorAll('.ant-select-item-option')).at(4)!);
+    expect(onSelect.mock.calls.length).toBe(2);
+
+    onSelect.mockReset();
+
+    const wrapper = render(
+      <Calendar
+        onSelect={onSelect}
+        defaultValue={Dayjs('2018-02-02')}
+        mode="year"
+        selectable={['date', 'year']}
+      />,
+    );
+
+    fireEvent.click(wrapper.container.querySelector('[title="2018-01"]')!);
+    fireEvent.click(wrapper.container.querySelector('[title="2018-03"]')!);
+    expect(onSelect.mock.calls.length).toBe(0);
+
+    onSelect.mockReset();
+
+    wrapper.rerender(<Calendar onSelect={onSelect} mode='month' selectable={['date', 'year']} />);
+
+    openSelect(wrapper.container, '.ant-picker-calendar-month-select');
+    fireEvent.click(
+      Array.from(wrapper.container.querySelectorAll('.ant-select-item-option')).at(5)!,
+    );
+    expect(onSelect.mock.calls.length).toBe(0);
+  });
+
+  it('only month chnage and year change should trigger onSelect callback', () => {
+    const onSelect = jest.fn();
+    const { container } = render(
+      <Calendar
+        onSelect={onSelect}
+        defaultValue={Dayjs('2018-02-02')}
+        selectable={['month', 'year']}
+      />,
+    );
+
+    fireEvent.click(container.querySelector('[title="2018-02-05"]')!);
+    fireEvent.click(container.querySelector('[title="2018-02-07"]')!);
+    expect(onSelect.mock.calls.length).toBe(0);
+
+    onSelect.mockReset();
+
+    openSelect(container, '.ant-picker-calendar-year-select');
+    fireEvent.click(Array.from(container.querySelectorAll('.ant-select-item-option')).at(2)!);
+    fireEvent.click(Array.from(container.querySelectorAll('.ant-select-item-option')).at(4)!);
+    expect(onSelect.mock.calls.length).toBe(2);
+
+    onSelect.mockReset();
+
+    const wrapper = render(
+      <Calendar
+        onSelect={onSelect}
+        defaultValue={Dayjs('2018-02-02')}
+        mode="year"
+        selectable={['month', 'year']}
+      />,
+    );
+
+    fireEvent.click(wrapper.container.querySelector('[title="2018-01"]')!);
+    fireEvent.click(wrapper.container.querySelector('[title="2018-03"]')!);
+    expect(onSelect.mock.calls.length).toBe(2);
+
+    onSelect.mockReset();
+
+    wrapper.rerender(<Calendar onSelect={onSelect} mode='month' selectable={['month', 'year']} />);
+
+    openSelect(wrapper.container, '.ant-picker-calendar-month-select');
+    fireEvent.click(
+      Array.from(wrapper.container.querySelectorAll('.ant-select-item-option')).at(5)!,
+    );
+    expect(onSelect.mock.calls.length).toBe(1);
+  });
+
```

Comment:
Should we add test case about `onChange`?

Reply:
I don't think it is necessary